server/services/cohorts.js

```diff
@@ -97,10 +97,21 @@ const getPSICohorts = async (psiID) => {
     });
 
   // calculate remaining cohort seats
-  psiCohorts = psiCohorts.map((cohort) => ({
-    ...cohort,
-    remaining_seats: cohort.cohort_size - cohort.participants.length,
-  }));
+  psiCohorts = psiCohorts.map((cohort) => {
+    const uniqueCohortParticipants = [];
+    // filter out duplicates
+    cohort.participants.forEach((participant) => {
```

Comment:
It looks like we should be able to alter the query to only join on current participants to fix this issue rather than an additional loop in JS

![image](https://user-images.githubusercontent.com/71518072/190291951-5eec06c7-9045-449f-ab33-45c14e73433b.png)

Reply:
Something like

```
.find({
  psi_id: psiID,
  // Additional condition to filter out the participants who don't have status assigned to them
  // This is used as a safeguard against legacy data
  // ref: https://github.com/bcgov/hcap/pull/847
  'participantStatusJoin.current !=': false,
+ 'participants.current': true,
  or: [
    { 'participantStatusJoin.status': ['hired', 'archived'] },
    { 'participantStatusJoin.status': null },
  ],
});
```

Reply:
Not exactly, this condition was present [before](https://github.com/bcgov/hcap/pull/847/commits/9e30b5b8cf07a6538c0aa4e14282fa21a74bd868#diff-eed45eb42a087a4e84846d8d5b06d6ce602ba0d8f8d26e6ae6a6550333c1ea95R64), but I had to remove it because the requirement is to display all participants including those who are no longer assigned to this cohort